repair/repair_task.hh

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (C) 2022-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#pragma once
+
+#include "task_manager/task_manager.hh"
+
+class repair_task : public task_manager::task::impl {
+public:
+    repair_task(task_manager::module_ptr module, task_id id, unsigned sequence_number = 0, std::string keyspace = "", std::string table = "", std::string type = "", std::string entity = "", task_manager::foreign_task_ptr&& parent = nullptr) 
+        : task_manager::task::impl(module, id, sequence_number, std::move(keyspace), std::move(table), std::move(type), std::move(entity), std::move(parent)) {
+        _status.progress_units = "SSTables";
+    }
+
+    future<task_manager::task::progress> get_progress() const override {
```

Comment:
This could be implemented in the generic task implementation.
The non-leaf tasks can return 0, 0 progress (could be implemented as the default virtual function for task::impl,
and the leaf tasks can return the actual progress.